components/Storyblok/SbChat.tsx

```diff
@@ -0,0 +1,66 @@
+import { storyblokEditable, StoryblokComponent, type SbBlokData } from '@storyblok/react/rsc';
+import { Chat } from '@/components/Chat';
+import { WidthBox, type WidthType } from '@/components/WidthBox';
+import { type MarginType } from '@/utilities/datasource';
+
+type SbChatProps = {
+  blok: {
+    _uid: string;
+    id: string;
+    boundingWidth?: 'site' | 'full';
+    width?: WidthType;
+    messages: SbBlokData[];
+    marginTop?: MarginType;
+    marginBottom?: MarginType;
+    isHidden?: boolean;
+  };
+};
+
+export const SbChat = ({
+  blok: {
+    id,
+    boundingWidth = 'full',
+    width,
+    messages,
+    marginTop,
+    marginBottom,
+    isHidden,
+  },
+  blok,
+}: SbChatProps) => {
+  if (isHidden || !messages.length) {
+    return null;
+  }
+
+  /**
+   * We calculate the delay for each message within a Chat session component and store the values in an array.
+   * The firs message has no delay (0 seconds).
+   * There is a base delay of 0.8 second between each subsequent message.
+   * If a message has the typing animation enabled (2 seconds),
+   * we add anotehr 2 second delay on top of the base 0.8 second to the next message.
+   */
+  const delays: number[] = [];
+  let cumulativeDelay = 0;
+
+  for (const message of messages) {
+    delays.push(cumulativeDelay);
+    cumulativeDelay += 0.8 + (message.showTyping ? 2 : 0);
+  }
```

Comment:
Instead of having the user enter the delay for each bubble manually, I have it calculated at the parent Chat component level. The idea is that there will be a default delay of 0.8 sec between each message. If they enable the typing animation for a bubble, that adds another 2 second delay. I've tested this on desktop and mobile and feels fairly natural to me.

Reply:
We aren't publishing stories with this component until March next year so there's plenty of time for editor feedback once they start using it.